### ISSUE TITLE
[Functions]Add support to specify Cardinality for EventHubTrigger

### ISSUE THREAD

**Issue by pragnagopa** [Microsoft/azure-maven-plugins#412]
### Plugin name and version
Functions
### Plugin configuration in your `pom.xml`
https://github.com/Azure/azure-functions-java-library/releases/tag/v1.0.0-beta-6 is now available in sonatype
```
https://oss.sonatype.org/service/local/staging/repository/commicrosoftazure-2642 
```

Example code to specify Cardinality 

```java

@FunctionName("EventHubTrigger")
    public void eventhubTrigger(
        @EventHubTrigger(name = "message", eventHubName = "%AzureWebJobsEventHubPath%", connection = "AzureWebJobsEventHubSender", cardinality = Cardinality.MANY) List<String> messages,
        final ExecutionContext context)
    {
        context.getLogger().info("Java Event Hub trigger received messages. Batch size: " + messages.size());
    }

```

should generate following function.json

```
"bindings" : [ {
    "type" : "eventHubTrigger",
    "name" : "message",
    "direction" : "in",
    "consumerGroup" : "$Default",
    "cardinality": "many",
    "connection" : "AzureWebJobsEventHubSender",
    "eventHubName" : "%AzureWebJobsEventHubPath%"
  } ],

```

Please note default is "many"

**Comment by pragnagopa** [comment]
@jdneo - Please take a look and let me know if you need anything else from me.

**Comment by Flanker32** [comment]
@pragnagopa  Can we use EventHubOutput to send event to Evnethub with cardinality=many? I write a HTTPTrigger which use EventHubOutput to trigger EventHubTrigger but it failed with error message
``` powershell
2018-10-12T09:26:32.483 [Error] Executed 'Functions.EventHubTriggerJava' (Failed, Id=adbc1993-7abd-4934-970d-7d55c83b0700)

Binding parameters to complex objects (such as 'Object') uses Json.NET serialization. 
1. Bind the parameter type as 'string' instead of 'Object' to get the raw values and avoid JSON deserialization, or
2. Change the queue payload to be valid json. The JSON parser failed: Unexpected character encountered while parsing value: H. Path '', line 0, position 0.
```
Here is the HTTP trigger 
``` java
 @FunctionName("HttpTrigger-Java")
    @EventHubOutput(name = "event", eventHubName = "flankerEventHub",dataType="string", connection = "flankerEventHubNamespace_RootManageSharedAccessKey_EVENTHUB")
    public String run(
            @HttpTrigger(name = "req", methods = {HttpMethod.GET, HttpMethod.POST}, authLevel = AuthorizationLevel.ANONYMOUS) HttpRequestMessage<Optional<String>> request,
            final ExecutionContext context) {
        return "HTTP TRIGGER";
    }
```
Besides , eventhub could work well when i send events through java api.

**Comment by pragnagopa** [comment]
No. Cardinality is for input/trigger only. We do no have support to write multiple values to one output binding.

**Comment by Flanker32** [comment]
@pragnagopa  I didn;t want to write multiple values to one output bindind, what i want to do is to test whether EventHubOutput could trigger EventHubTrigger with cardinality=many, i write a test project with HTTP Trigger and EventHubTrigger, HTTPTigger will write message to EventHub with EventHubOutput, but it break down when EventHubTrigger execute its function, so is there any way to solve this issue?

**Comment by pragnagopa** [comment]
You need to bind to a list as setting cardinality to many will enable receiving a set of message.Please see sample code here https://github.com/Azure/azure-functions-java-worker/blob/dev/endtoendtests/src/main/java/com/microsoft/azure/functions/endtoend/EventHubTriggerTests.java

**Comment by Flanker32** [comment]
@pragnagopa  I tried again but the issue remains, here is the trigger of my project, is there anything wrong in this code?
``` java
@FunctionName("EventHubTriggerJava")
    public void run(
            @EventHubTrigger(name = "message", eventHubName = "flankerEventHub", connection = "flankerEventHubNamespace_RootManageSharedAccessKey_EVENTHUB", consumerGroup = "$Default", cardinality = Cardinality.MANY) String[] message,
            final ExecutionContext context) {
        context.getLogger().info("Java Event Hub trigger function executed.");
        context.getLogger().info("Length:" + message.length);
        Arrays.stream(message).forEach(singleMessage -> context.getLogger().info(singleMessage));
    }
```

``` java
    @FunctionName("HttpTrigger-Jav-Single")
    public void run(
            @HttpTrigger(name = "req", methods = {HttpMethod.GET, HttpMethod.POST}, authLevel = AuthorizationLevel.ANONYMOUS) HttpRequestMessage<Optional<String>> request,
            @EventHubOutput(name = "output", eventHubName = "flankerEventHub", connection = "flankerEventHubNamespace_RootManageSharedAccessKey_EVENTHUB") OutputBinding<String> output,
            final ExecutionContext context) {
                output.setValue("SINGLETRIGGER");
    }
```

**Comment by pragnagopa** [comment]
Yes. Use List<String> instead of String[]

Thanks,
Pragna
________________________________

**Comment by pragnagopa** [comment]
Also set datatype to String if your test input is not json

Thanks,
Pragna
________________________________

**Comment by Flanker32** [comment]
Solved in #423
